app/models/works_generator.rb

```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+class WorksGenerator
+  def initialize(webaccess_id)
+    raise ArgumentError, 'No Webaccess ID supplied' if webaccess_id.blank?
+
+    @user = User.find_by(webaccess_id: webaccess_id)
```

Comment:
Maybe this should raise an exception if it can't find that user? Or `find_or_create_by`?